src/tests/unit/tests/common/get-card-selection-view-data.test.ts

```diff
@@ -29,109 +38,228 @@ describe('getCardSelectionStoreviewData', () => {
             focusedResultUid: null,
         };
 
-        initialState = cloneDeep(defaultState);
+        initialUnifiedScanResultState = {
+            platformInfo: {
+                viewPortInfo: {},
+            },
+            results: [
+                {
+                    uid: 'sampleUid1',
+                    status: 'fail',
+                },
+                {
+                    uid: 'sampleUid2',
+                    status: 'fail',
+                },
+                {
+                    uid: 'sampleUid3',
+                    status: 'fail',
+                },
+                {
+                    uid: 'sampleUid4',
+                    status: 'fail',
+                },
+                {
+                    uid: 'sampleUid5',
+                    status: 'pass',
+                },
+            ],
+        } as UnifiedScanResultStoreData;
+
+        getUnavailableHighlightStatus = Mock.ofType<GetUnavailableHighlightStatus>();
+        initialCardSelectionState = cloneDeep(defaultCardSelectionState);
     });
 
     test('all rules collapsed, visual helper enabled, expect all highlights', () => {
-        const viewData = getCardSelectionViewData(initialState);
-
-        expect(viewData.highlightedResultUids).toEqual([
-            'sampleUid1',
-            'sampleUid2',
-            'sampleUid3',
-            'sampleUid4',
-        ]);
+        const viewData = getCardSelectionViewData(
+            initialCardSelectionState,
+            initialUnifiedScanResultState,
+            getUnavailableHighlightStatus.object,
+        );
+
+        expect(viewData.resultsHighlightStatus).toEqual({
+            sampleUid1: 'visible',
+            sampleUid2: 'visible',
+            sampleUid3: 'visible',
+            sampleUid4: 'visible',
+        });
+        expect(viewData.expandedRuleIds).toEqual([]);
+        expect(viewData.selectedResultUids).toEqual([]);
+        expect(viewData.visualHelperEnabled).toEqual(true);
+    });
+
+    test('all rules collapsed, visual helper enabled, some results unavailable', () => {
```

Comment:
It would be good to add test cases that verify the precedence of `unavailable` vs `hidden` in the "some rules expanded" and "some rules selected" states